### DISCUSSION TITLE
Berechnung der AZ durch Vaillant

### DISCUSSION THREAD

**Discussion by Apolyxon** [cyberthom42/vaillant-arotherm-plus#2] [General]
Da wir gerne das Mysterium lösen wollen, wie Vaillant die interne AZ berechnet, benötigen wir dazu einige Daten, die @cyberthom42 hochgeladen und ins Repository gestellt hat.

Der Thread dient zur Diskussion der Berechnung und eventueller Ergebnisse.

**Comment by Apolyxon** [comment]
@cyberthom42 

Erste Ergebnisse

Enteisung wird als negativer Wert automatisch berücksichtigt.

AZ Monat berechnet aus Durchfluss, VL, RL mitsamt angezeigtem Strom: 

Heizen: 1557 Heizertrag / 403 Stromverbrauch = AZ Dezember 1-22:  3,86
WW: 159 / 60 = 2,65

Da ich die Informationen noch nicht zur Vaillant-AZ sehe, kann ich leider noch nichts dazu sagen, ob die Berechnung passen kann. Da Enteisungen nur wenige Minuten dauern, ist aber eine höhere Auflösung (30 Sek) sicher nicht verkehrt!

**Comment by Apolyxon** [comment]
PS: Stromverbrauch Real AZ wäre

Heizen: 1557 / 454 =  3,42

159 / 63 = 2,52

**Comment by cyberthom42** [comment]
Ich habe eine neue Tabelle hochgeladen. Die Energiedaten, die die Hydraulikstation anzeigt sind schon einmal identisch zu den Daten die über ebus geloggt werden. 
Allerdings weichen die Energiedaten (Verbrauch) in der MyVaillant App leicht von diesen Daten ab. Die MyVaillant App zeigt leicht höhere Verbrauchswerte an. Die Kälteenergie stimmt aber bei beiden. Die angezeigte Arbeitszahl / COP in der App passt zur Berechnung Wärmeenergie/Stromverbrauch. Die Hydraulikstation zeigt jedoch eine viel niedrigere AZ an. Im Dezember ist das bei Heizen AZ 3,8 in der App und 3,5 auf der Hydraulikstation.
Der COP, der von der Hydraulikstation angezeigt wird ist definitiv falsch. Das ist mir bis dato gar nicht so aufgefallen. Es ist auch nicht sofort ersichtlich, da die Hydraulikstation nicht den monatlichen Stromverbrauch angibt.

**Reply by fred081273** [reply]
Ihr schreibt hier von COP und angezeigter AZ. Wo genau ist da bei der Hydraulikstation der Unterschied? Ich habe ja eher niedrige Kälteleistungen, der Verbrauch ist aber ganz ok. Daher wäre es wichtig zu wissen, ob die von der Hydraulikstation angezeigten Kälteleistungen passen.
Meine schlechten AZ möchte ich gerne bei Vaillant adressieren. Beim Heizen habe ich im Dezember z.B. nur 2,8. Dafür wäre es hilfreich wenn ich wüsste, ob diese Werte von der Hydraulikstation halbwegs passen. Danke Euch!

**Reply by cyberthom42** [reply]
Den COP berechnen wir, die Arbeitszahl wird von der Hydraulikstation angegeben. Vaillant dokumentiert nicht, was in der angezeigten Monats Arbeitszahl alle mit eingerechnet wird. Das Ziel ist zu sehen, was bei der Anzeige der Arbeitszahl mit berücksichtigt wird und wie genau die Werte sind.

**Comment by Apolyxon** [comment]
Der Stromverbrauch real scheint zusammen mit VL/RL, Durchfluss als Heizleistung den angezeigten Wert gut zu tracken!

Überrascht uns Vaillant? Ich wüsste ehrlich gesagt nicht wie ich die JAZ groß besser berechnen sollte. Kann es sein, dass die AZ Anzeige der Hydraulikstation tatsächlich echt gut ist? Sogar hier besser als die App, die eher den ungenauen "Verbrauch_anzeige" nutzt.

<img width="1565" alt="image" src="https://user-images.githubusercontent.com/3678079/209180629-c2bb026a-0045-48f2-9b42-81f627b53a8e.png">

**Comment by Apolyxon** [comment]
Mit einem Offset von 18W (oder ähnliche Werte vom real gemessenen Strom), erhalte ich für die Heizung ein perfektes Tracking.

Ich würde also 2 Dinge festhalten:

1.  Es werden keine komischen "gerundeten" Werte von der Hydraulikstation herangezogen, sondern gute und Wattgenaue Messwerte genommen
2. Einige fixe Verbraucher werden nicht berücksichtigt - dies sagt Vaillant aber auch selbst. Die scheinen ca. 20 Watt im Schnitt auszumachen.

<img width="178" alt="image" src="https://user-images.githubusercontent.com/3678079/209309679-b2545d51-ca69-428c-92da-32f6831e5ad2.png">


<img width="728" alt="image" src="https://user-images.githubusercontent.com/3678079/209309460-d8592a42-abc4-4837-98cf-3b661ba01093.png">

**Reply by cyberthom42** [reply]
Wie hast du JAZ berechnet, vor allem welche Werte hast du als Grundlage genommen? 
Interessant wäre nun noch der Vergleich, warum die COP Werte in der App so abweichen.

**Reply by Apolyxon** [reply]
Ich habe dafür einerseits versucht (bei Enteisungen etwas ungenau da die nur kurz dauern) die aktuelle Heizleistung zu ermitteln (siehe Wiki). Außerdem habe ich den Stromverbrauch für das Heizen ermittelt, indem ich den Stromverbrauch (real) genommen habe und diesen aufsummiere. Beim Aufsummieren habe ich dann (manuell ermittelt) 18 Watt abgezogen. Dies indem ich folgendes gemacht habe:

Strom_aktuell in kWMin
=WENN([@Heizungspumpe]=1;WENN([@[Stromv_Anzeige]] > 0; [@[Stromv_gemessen]] - 0.018;0);0)/60

Das kann ich dann aufsummieren und erhalte den Stromverbrauch (nach 60 Einträgen = 1 h) in kWh.

Wenn ich 12 Watt nehme, passt es auch super.

**Comment by cyberthom42** [comment]
Super Arbeit. Wie bist du auf 18 Watt gekommen? Aktuell messe ich den realen Verbrauch der gesamten Anlage (arotherm, Hydraulikstation, Frischwasserstation). Der Standby Verbrauch liegt bei 13 Watt. Könnte also ganz gut passen. 
Ich könnte mal noch den Verbrauch der WW Pumpe und der Zirkulationspumpe messen. Sieht man aber auch an den Statistiken, da ich Zirkulation und WW-Durchfluss auch mit tracke. Ich sehe schon, am besten wäre wahrscheinlich wenn ich einen export der InfluxDB hochlade, oder ?
Der angegebene Stromverbrauch ist die Sumem der 3 Phasen. Die arotherm hängt ja nur an einer Phase. Ich logge auch den Verbrauch jeder Phase separat. Ich befürchte aber, dass die Frischwasserstation an der gleichen Phase hängt. Die Hydraulikstation ist 3-phasig angeschlossen. Ich werde das noch mal prüfen. Dann könnten wir die Theorie dann noch bestätigen.

**Comment by Apolyxon** [comment]
13 Watt passt auch super.

<img width="788" alt="image" src="https://user-images.githubusercontent.com/3678079/209356200-402f6755-0978-43fd-885c-d27cb2a35a38.png">

**Comment by Apolyxon** [comment]
Zum 31.12 dann nochmal den gesamten Monat bereitstellen. Wird nochmal spannend mit der jetzigen AZ Steigerung aufgrund der Temperatur.

Ansonsten versuche ich aktuell mal rauszubekommen, wie Vaillant den Stromverbrauch (Heizen) berechnet. Der ist offenbar ohne Enteisungen... Es war ja aufgefallen, dass (Energieertrag + Stromverbrauch) / Stromverbrauch nicht die berechnete AZ ist

**Reply by Apolyxon** [reply]
Den Stromverbrauch (Heizen) laut Hydraulikstation kriege ich sehr genau getrackt mit folgender Formel:

=WENN(UND([@Heizungspumpe]=1;[@[4-Wege-V]]="off");WENN([@[Stromv_Anzeige]]>0; [@[Stromv_gemessen]]-0.025;0);0)/60

Da ist selbst nach 20 Tagen ein Fehler von unter 1 %. Die 25 Watt sind natürlich noch eine recht wackelige Konstante. Der Stromverbrauch ist hierfür ganz offenbar **ohne** Enteisungen!

Ich vermute, dass aber die Enteisungen für die AZ berücksichtigt werden. Das ist hier sehr eindeutig zu sehen. Die Berechnung der AZ anhand des Vaillant-"Stromverbrauchs" in Summe und des "Energieertrags Heizen" ist daher nutzlos.

<img width="191" alt="image" src="https://user-images.githubusercontent.com/3678079/209360569-709d4bde-9b57-4d37-9c50-3172aba1ed83.png">

**Reply by Apolyxon** [reply]
@cyberthom42

**Reply by cyberthom42** [reply]
Cool. Ich versuche dann mal ab 1.1. die Phase der WP umzuklemmen, dann kann ich auch den Verbrauch zwischen Hydraulikstation / Frischwasserstation und WP vergleichen. Interessant ist hierbei wirklich, dass die Anzeige des Stromverbrauches nur sehr grob gerundet zu sein scheint, intern aber ein genauer Wert vorliegt.

**Reply by Apolyxon** [reply]
@cyberthom42 Lad mal die Daten bis 31.12 Mitternacht hoch :)

**Reply by cyberthom42** [reply]
Habe ich hochgeladen

**Comment by Apolyxon** [comment]
<img width="682" alt="image" src="https://user-images.githubusercontent.com/3678079/210821176-81f41207-f1e3-408f-b060-833013e20c83.png">

q.e.d.

Heizleistung
`=WENN([@Heizungspumpe]="1";([@VL]-[@RL])*1.16*[@Durchfluss];0)/60000`

Stromverbrauch
`=WENN([@Heizungspumpe]="1";WENN([@[Stromv_Anzeige]] > 0; [@[Stromv_gemessen]] - 0.018;0);0)/60`

Summe Heizleistung / Strom
`=SUMME(..)`

JAZ Real
`=[@[Summe Heizleistung]]/[@[Summe Strom]]`

**Reply by Apolyxon** [reply]
Warmwasser

<img width="700" alt="image" src="https://user-images.githubusercontent.com/3678079/210822142-934b8360-7505-4fba-9816-3e74ef8c1530.png">

**Reply by cyberthom42** [reply]
Das sieht ja ziemlich genau aus. Ich bin gerade dran zu prüfen, wo der Standby Verbrauch her kommt. Die Frischwasserstation hatte ich in verdacht, die brauch aber nur 2,2W. Ich werde die Tage etwas weniger aktiv sein, wir haben einen Trauerfall.

**Reply by Apolyxon** [reply]
Mein Beileid.

Ja die JAZ der Hydraulikstation scheint gut gemacht zu sein.

**Reply by Apolyxon** [reply]
<img width="519" alt="image" src="https://user-images.githubusercontent.com/3678079/210827292-fa8da600-41fc-494e-b2a6-ca2a15ad43e8.png">

Stromverbrauch ohne Enteisungen (Heizen)  passt auch gut (hier mit 25 W Offset)

Hast du irgendwo in der DB den Wert für Stromverbrauch (Gesamt) laut Vaillant? Also nicht den Monatlichen?

**Reply by cyberthom42** [reply]
Ja, aber getrennt nach Heizen (PrEnergySumHc) und Warmwasser. Der ebus-Wert für Gesamt lässt sich nicht auslesen, hier passen die HEX-Werte in der CSV noch nicht.
Ich habe also Energieverbrauch WW gesamt und pro monat, das gleiche auch für Heizung. Beides mit dem gleichen Intervallen in die DB geloggt. Wenn das hilft, kann ich das CSV noch aktualisieren

**Comment by Apolyxon** [comment]
@cyberthom42 Der Januar war ja temperaturtechnisch auch hochspannend. Kannst du da mal bis heute die Daten hochladen damit wir da einen 2. Monat haben zur Verifikation?

**Reply by Apolyxon** [reply]
Falls du hast: Die AZ der App zu diversen Zeitpunkten wäre auch top - wenigstens aber heute. Vielleicht hast du da ja zufällig den einen oder anderen Wert notiert.

**Comment by cyberthom42** [comment]
Die 13 Watt passen übrigens ganz gut. Wenn ich die WP über das Menü am 720 "Heizen" "aus" stelle habe ich einen Gesamtstromverbrauch von 14,4W
